src/colony/modules/net.js

```diff
@@ -163,7 +147,17 @@ TCPSocket.prototype.connect = function (/*options | [port], [host], [cb]*/) {
 
   function setUpConnection(ip) {
     if (self.socket == null) {
-      if (self._secure) ensureSSLCtx();
+      if (self._secure) {
+        var custom_certs = null;
+        self._ssl_checkCerts = (opts.rejectUnauthorized !== false);
```

Comment:
Heh, `opts` is (correctly) assumed to be non-`null` the very line after it gets assigned:

```
  var args = normalizeConnectArgs(arguments);
  var opts = args[0];
  if (opts.allowHalfOpen) console.warn("Ignoring allowHalfOpen option.");
```